### PR TITLE
Small changes in many files

### PR DESCRIPTION
**Pull request by brickviking**
Includes four sets of patches, including removing redundant "the", and a comment I need clarification on what it was meant to include but left out. More importantly (and I don't know why Dolphin isn't falling over) is the update from dolphindr7.dll to dolphindr8.dll in about 58 resource_Default_view methods. Is there a fallback method so these methods don't tie themselves in knots?

One other thing, I've found in the defaultResourceLibrary at Core/Object Arts/Dolphin/Base/Core.SessionManager.cls:247
```
defaultResourceLibrary
	"Private - Answer an `External.ResourceLibrary` containing the Dolphin development system resources."

	resourceLibrary isNil
		ifTrue: 
			[resourceLibrary := External.ResourceLibrary
						open: self defaultResLibPath
						withAliases: #('dolphindr7' 'dolphindr006' 'dolphindr005' 'dolphindr004' 'dolphindevres' 'dolphindr993')
						flags: 0].
	^resourceLibrary!
```
Should the withAliases: have 'dolphindr8' added to that list, or does Dolphin already know about that file?

Again, an eye over these to see if I missed anything obvious. Thanks for the work you're doing, Blair.

### CODE REVIEW

File: Core/Object Arts/Dolphin/MVP/Presenters/Prompters/UI.IntegerPrompter.cls [hunk 1]
```diff
@@ -84,7 +84,7 @@ on: aValueModel prompt: aStringPrompt caption: aStringCaptionOrNil
 		interval: (1 to: SmallInteger maximum)!
 
 on: aValueModel prompt: aStringPrompt caption: aStringCaptionOrNil interval: anInterval 
-	"Creates an instance of the receiver on aValueModel with and aStringPrompt. 
+	"Creates an instance of the receiver on aValueModel with an aStringPrompt.
```
Comment (blairmcg) [inline Core/Object Arts/Dolphin/MVP/Presenters/Prompters/UI.IntegerPrompter.cls:87]: Creates an instance of the receiver on aValueModel and aStringPrompt.

File: Core/Object Arts/Dolphin/MVP/Presenters/Text/OS.CHARFORMAT2W.cls [hunk 2]
```diff
@@ -303,12 +303,12 @@ logFont
...
 
 maskAny: anInteger
-	"Private - Answer true if the mask member contains the bit flags represented by anInteger."
+	"Private - Answer true if the mask field contains the bit flags represented by anInteger."
 
 	^self dwMask anyMask: anInteger!
 
 maskIn: anInteger
-	"Private - OR in to contents mask member the bit flags represented by anInteger.
+	"Private - OR in to contents mask field the bit flags represented by anInteger.
```
Comment (blairmcg) [inline Core/Object Arts/Dolphin/MVP/Presenters/Text/OS.CHARFORMAT2W.cls:311]: to _the_ contents mask

File: Core/Object Arts/Dolphin/MVP/Presenters/Text/OS.CHARFORMAT2W.cls [hunk 3]
```diff
@@ -318,12 +318,12 @@ maskOut: anInteger
 
 pointSize
 	"Answer the receiver's height field in points (N.B. the answer may be
-	a Fraction if the, for example, the size if 10.5 points)."
+	a Fraction if the, for example, the size is 10.5 points)."
```
Comment (blairmcg) [inline Core/Object Arts/Dolphin/MVP/Presenters/Text/OS.CHARFORMAT2W.cls:321]: Extraneous the after "a Fraction if"

File: Core/Object Arts/Dolphin/MVP/Presenters/Text/OS.PARAFORMAT.cls [hunk 4]
```diff
@@ -89,7 +89,7 @@ dxStartIndent: anInteger
 	bytes sdwordAtOffset: _OffsetOf_dxStartIndent put: anInteger!
 
 maskIn: anInteger
-	"Private - OR in to contents mask member the bit flags represented by anInteger.
+	"Private - OR in to receiver's mask field the bit flags represented by anInteger.
```
Comment (blairmcg) [inline Core/Object Arts/Dolphin/MVP/Presenters/Text/OS.PARAFORMAT.cls:92]: OR in to _the_ receiver's

File: Core/Object Arts/Dolphin/MVP/Presenters/Text/UI.MultilineTextEdit.cls [hunk 5]
```diff
@@ -87,7 +87,7 @@ currentLine
 !
 
 cutLine
-	"Delete the line on which the caret is located, including the line delimiter (i.e. subsequent lines shift
+	"Delete the line on which the caret is located into the cut buffer, including the line delimiter (i.e. subsequent lines shift
```
Comment (blairmcg) [inline Core/Object Arts/Dolphin/MVP/Presenters/Text/UI.MultilineTextEdit.cls:90]: I think I would say "to the clipboard"

File: Core/Object Arts/Dolphin/MVP/Presenters/Text/UI.RichTextEdit.cls [hunk 6]
```diff
@@ -614,7 +614,7 @@ selectionCharFormat
 
 selectionCharFormat: aCHARFORMAT 
 	"Private - Apply the specified character formatting options to the receiver's
-	current selection (or from the insertion point if no selection."
+	current selection (or from the insertion point if no selection.)"
```
Comment (blairmcg) [inline Core/Object Arts/Dolphin/MVP/Presenters/Text/UI.RichTextEdit.cls:617]: no selection).

File: Core/Object Arts/Dolphin/MVP/Presenters/Text/UI.TextEdit.cls [hunk 7]
```diff
@@ -733,7 +733,7 @@ onInputMaskRequired: anEvent
 	WM_GETDLGCODE to ensure the control never actually receives it. This keeps 
 	its behaviour sane, and means we do not need workarounds in WM_CHAR (to 
 	forcibly insert VK_TABs). It also enables us to implement WantTab functionality
-	which windows does not support directly as an Edit control style."
+	which Windows does not support directly as an Edit control style."
```
Comment (blairmcg) [inline Core/Object Arts/Dolphin/MVP/Presenters/Text/UI.TextEdit.cls:736]: "functionality, which"
(or without the comma, "functionality that", as you prefer)

File: Core/Object Arts/Dolphin/MVP/Presenters/Tree/UI.TreePresenter.cls [hunk 8]
```diff
@@ -9,7 +9,7 @@ UI.SelectableItemsPresenter subclass: #'UI.TreePresenter'
 UI.TreePresenter guid: (Core.GUID fromString: '{87b4c688-026e-11d3-9fd7-00a0cc3e4a32}')!
 UI.TreePresenter comment: 'TreePresenter implements a <treePresenter> component that can be used to display the contents of a <treeModel> within a <treeView>. This is similar to the way a tree view is used to display a file system''s contents in the Window''s explorer. For presentation of a tree as a hierarchy of nested "folders" see the <FolderPresenter> class.
 
-TreePresenter conforms to <selectableItems> to provide methods for accessing the current selection. In addition, the <treePresenter> protocol allows the tree to be navigating using the #expand: , #collapse: and #up methods.
+TreePresenter conforms to <selectableItems> to provide methods for accessing the current selection. In addition, the <treePresenter> protocol allows the tree presentation to be modified using the #expand: , #collapse: and #up methods.
```
Comment (blairmcg) [inline Core/Object Arts/Dolphin/MVP/Presenters/Tree/UI.TreePresenter.cls:12]: I think the original text is more correct if "navigating" is replaced with "navigated"

File: Core/Object Arts/Dolphin/MVP/Views/Scintilla/UI.Scintilla.ScintillaView.cls [hunk 9]
```diff
@@ -9827,7 +9827,7 @@ xOffset: xOffsetInteger
 		lParam: 0!
 
 zoomFactor
-	"Scintilla zooms in and out by points, and is the the points delta that is reported by SCI_GETZOOM, which is not easy to understand from a user perspective. The actual the zoom factor is relative to the original font size. The factor can be multiplied by 100 to give a zoom percentage for presentation to the user."
+	"Scintilla zooms in and out by points, and is the points delta that is reported by SCI_GETZOOM, which is not easy to understand from a user perspective. The actual the zoom factor is relative to the original font size. The factor can be multiplied by 100 to give a zoom percentage for presentation to the user."
```
Comment (blairmcg) [inline Core/Object Arts/Dolphin/MVP/Views/Scintilla/UI.Scintilla.ScintillaView.cls:9830]: and _it_ is the points

Comment (blairmcg) [reply Core/Object Arts/Dolphin/MVP/Views/Scintilla/UI.Scintilla.ScintillaView.cls:9830]: "The actual ~~the~~ zoom factor"

File: Core/Contributions/Camp Smalltalk/ANSI/Tests/AnsiDB.st [hunk 10]
```diff
@@ -2459,7 +2459,7 @@ ProtocolSpec newMessagePattern: 'year: hour day: dayOfYear hour: minute minute:
...
+It is possible that the time specified does not exist in the local time specified by the implementation. If there is a time change such that the local time is set forward and the time specified is in the interregnum, then that time does not exist in the local time. For example if at 02:00 in California on April 26, 1997 there is a time change that sets local time forward one hour, then the local time 02:30  in California does not exist. Conversely if there is a time change that sets the locale time back there are times which are ambiguous. For example if instead of setting the local time forward from 02:00 to 03:00 it is set back to 01:00 the local time 01:30 in California is ambiguious. The result is the least <DateAndTime> that conforms to the given parameters.
```
Comment (blairmcg) [inline Core/Contributions/Camp Smalltalk/ANSI/Tests/AnsiDB.st:2462]: In some cases 'the the' is not a doubling up, but rather a mistyping of 'then the'. That is the case here. It either needs the 'then', or a comma before 'the'. I'm not sure why this paragraph is repeated so much, but disregarding that the same would apply to the duplicates too.

**Review by blairmcg — CHANGES_REQUESTED** [review]

File: Core/Object Arts/Dolphin/MVP/Base/Graphics.Icon.cls [hunk 11]
```diff
@@ -342,6 +342,7 @@ asterisk
 
 blank
 	Blank isNil ifTrue: [Blank := self fromId: 'Blank.ico'].
+	NtDll isWine ifTrue: [^self null].
```
[the remark this answers is not in the document]
Comment (brickviking) [reply Core/Object Arts/Dolphin/MVP/Base/Graphics.Icon.cls:345]: Ah, sorry. That one is an attempt to fix the black squares I was getting once you had upscaled Blank.ico to fix issue #1097. Windows seems to downscale the image fine, but for some reason I don't think wine knows quite what to do. I know you don't want to litter the image with a whole lot of useless platform-specific fixes but I was getting the same black squares that #1097 had, though for a different reason. Do you have a better suggestion that would remove the black squares?

My bug report is linked [here in my issue 23](https://github.com/brickviking/Dolphin/issues/23) to keep it out of the flow of general Dolphin bugfixing.

File: Core/Object Arts/Dolphin/MVP/Presenters/ListTree/UI.ListTreeView.cls [hunk 12]
```diff
@@ -30,7 +30,7 @@ There are some examples of use in the class-side category ''examples'', but to s
 
 Instances can be used with TreePresenters (e.g. TreePresenter.ListTree view), ListPresenters (e.g. ListPresenter.ListTree view), or with the specialised presenter class, TreeViewPresenter, which knows more about ListTreeView than the pre-packaged Presenter classes do.  
 
-The multi-select version, MultipleSelectionListTreeView, can be used with ListPresenters (e.g. ListPresente.Multi-select ListTree view) and with ProperTreeViewPresenter.
+The multi-select version, MultipleSelectionListTreeView, can be used with ListPresenters (e.g. ListPresenter.Multi-select ListTree view) and with ProperTreeViewPresenter.
```
[the remark this answers is not in the document]
Comment (brickviking) [reply Core/Object Arts/Dolphin/MVP/Presenters/ListTree/UI.ListTreeView.cls:33]: Down a bit later in the comment (around line 71 before removals), there's mention made of multiple selections in a ListTreeView. Does the primary ListTreeView control support multiple selections as well as single selections?

File: Core/Object Arts/Dolphin/MVP/Presenters/ListTree/UI.ListTreeView.cls [hunk 13]
```diff
@@ -68,11 +68,11 @@ Differences between ListTreeView and TreeView
 
 The biggest deficiency (that I know about so far) is that ListTreeView does not draw tree ''lines''.  Since M$ don''t do this either (in Outlook Express) I take it that this is OK.  If you find it too much of a failing, then don''t use this software... ;-)
 
-The big gains over a TreeView is that a ListTreeView supports additional ListViewColumns and that MultipleSelectionListTreeView supports multiple selection (duh).  As far as I know things like per-column images, owner-draw, etc all work, but I haven''t tried everything.
+The big gains over a TreeView is that a ListTreeView supports additional ListViewColumns and that MultipleSelectionListTreeView supports multiple selection (duh).  As far as I know things like per-column images, owner-draw, etc. all work, but I haven''t tried everything.
```
[the remark this answers is not in the document]
Comment (brickviking) [reply Core/Object Arts/Dolphin/MVP/Presenters/ListTree/UI.ListTreeView.cls:71]: Right. Do I leave in the phrase "As far as I know things like ..."?

File: Core/Object Arts/Dolphin/MVP/Presenters/ListTree/UI.ListTreeView.cls [hunk 14]
```diff
@@ -160,8 +160,8 @@ allExpandedItems
 	^ OrderedCollection withAll: expanded.!
 
 allHiddenExpandedItems
-	"answer an OrderedCollection of all the items that are currently hidden-expanded; that
-	is the ones that we will automatically expand as soon as they are visible, because they
+	"answer an OrderedCollection of all the items that are currently hidden-expanded; those
```
[the remark this answers is not in the document]
Comment (brickviking) [reply Core/Object Arts/Dolphin/MVP/Presenters/ListTree/UI.ListTreeView.cls:163]: Perhaps we could use the following:
```
answer an OrderedCollection of all the items that are currently hidden-expanded. These are 
the ones we will automatically expand as soon as they are visible, ...
```

File: Core/Contributions/IDB/ChunkBrowser.cls [hunk 15]
```diff
@@ -501,7 +501,7 @@ resource_Default_view
...
...8 1 2530 #helpWhatsThis 8 'What''s This?' 5345 1 nil nil nil 2626 4097 2498 1 2530 #learnSmalltalk 8 'Learn Smalltalk' 1 1 1378630 2 ##(Smalltalk.TextTileIcon) $\x1F393 nil 626 #orchid 5040 nil nil 1 nil nil nil nil 2626 4097 2498 1 2530 #helpWhatsNew 8 'What''s &New' 1 1 nil nil nil 2498 1 2530 #helpGuidedTour 8 '&Guided Tour' 1 1 nil nil nil 2498 1 2530 #helpTutorials 8 'Tutorials' 1 1 nil nil nil 2626 4097 2498 1 2530 #objectArtsHomePage 8 'Object Arts Homepage' 1 1 9538 nil true 9584 8 '!!APPLICATION' 9632 nil nil 2498 1 2530 #dolphinNewsgroup 8 'Dolphin Newsgroup/Forum' 1 1 nil nil nil 2498 1 2530 #dolphinWikiWeb 8 'Dolphin WikiWeb' 1 1 nil nil nil 2626 4097 2498 1 2530 #aboutDolphin 8 '&About Dolphin Smalltalk' 1 1 9826 $\x1F42C nil nil 626 #darkSlateBlue nil nil 1 nil nil nil nil 8 '&Help' #help 134217729 nil nil 27829 nil nil 8 '' nil 1 nil nil nil nil nil nil nil nil 1 nil nil nil nil 1 nil nil 802 138 144 34 2 866 #createAt:extent: 34 2 498 6143 21 498 1361 1001 416 866 #text: 34 1 8 'ChunkBrowser' 416 946 8 #[44 0 0 0 0 0 0 0 0 0 0 0 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 255 11 0 0 10 0 0 0 167 14 0 0 254 1 0 0] 34 3 560 2160 1520 1008 nil 27 )!
```
[the remark this answers is not in the document]
Comment (brickviking) [reply Core/Contributions/IDB/ChunkBrowser.cls]: Yup, fair call. I've undone those changes. Are those numbers mainly raw Smalltalk commands to be interpreted that I see in the method body?

File: Core/Object Arts/Dolphin/MVP/Views/Scintilla/UI.Scintilla.ScintillaView.cls (hunk 9 above)
Comment (brickviking) [reply Core/Object Arts/Dolphin/MVP/Views/Scintilla/UI.Scintilla.ScintillaView.cls:9830]: Also, how about this?

```
"Scintilla zooms in and out by points, and reports the points delta with SCI_GETZOOM, which is not easy to understand from a user perspective.
```

**Comment by blairmcg** [comment]
> One other thing, I've found in the defaultResourceLibrary at Core/Object Arts/Dolphin/Base/Core.SessionManager.cls:247
> ...
> Should the withAliases: have 'dolphindr8' added to that list, or does Dolphin already know about that file?

I forgot to answer that. 

No, the list of aliases is things that should map to the lastest resource library if requested instead of it. If you look at `defaultResLibPath` you will see that it returns `DolphinDR8`.

File: Core/Contributions/IDB/ChunkBrowser.cls (hunk 15 above)
[the remark this answers is not in the document]
Comment (blairmcg) [reply Core/Contributions/IDB/ChunkBrowser.cls]: They are a representation of serialized objects using only literal values (strings, symbols, ints, booleans, nil, byte arrays, arrays of other literals, and so on). Not so much commands, as data.

File: Core/Object Arts/Dolphin/MVP/Base/Graphics.Icon.cls (hunk 11 above)
[the remark this answers is not in the document]
Comment (blairmcg) [reply Core/Object Arts/Dolphin/MVP/Base/Graphics.Icon.cls:345]: We should deal with that separately. You are right that I don't want to have any of these `isWine` conditions. In most cases it should be possible to work around in another way, although equally I don't want lowest common denominator solutions either as that would be outside the philosophy of what Dolphin was originally intended to be (i.e. a native Windows Smalltalk).

File: Core/Object Arts/Dolphin/MVP/Presenters/ListTree/UI.ListTreeView.cls (hunk 13 above)
[the remark this answers is not in the document]
Comment (blairmcg) [reply Core/Object Arts/Dolphin/MVP/Presenters/ListTree/UI.ListTreeView.cls]: Yes. This code and the comments were originally written by Chris Uppal. I don't think we should change his comments unless they are inaccurate.

File: Core/Object Arts/Dolphin/MVP/Presenters/ListTree/UI.ListTreeView.cls (hunk 14 above)
[the remark this answers is not in the document]
Comment (blairmcg) [reply Core/Object Arts/Dolphin/MVP/Presenters/ListTree/UI.ListTreeView.cls:160]: Again, I don't feel it is appropriate to change the writing style in contributions. It forms part of the contribution.

File: Core/Object Arts/Dolphin/MVP/Base/Graphics.Icon.cls (hunk 11 above)
[the remark this answers is not in the document]
Comment (brickviking) [reply Core/Object Arts/Dolphin/MVP/Base/Graphics.Icon.cls:345]: Okay, I'll drop it out of the PR. But man those black squares look ugly. :)
Makes me wonder what stuff Jgfoster did to get his image to work, it appears to be a 99.5% vanilla Dolphin.

**Pull request merged**